src/react-components/ui-root.js

```diff
@@ -0,0 +1,422 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import NameEntryPanel from './name-entry-panel';
+import { VR_DEVICE_AVAILABILITY } from "../utils/vr-caps-detect";
+import queryString from "query-string";
+import { SCHEMA } from "../storage/store";
+const { detect } = require("detect-browser");
+
+const browser = detect();
+
+const ENTRY_STEPS = {
+  start: "start",
+  mic_grant: "mic_grant",
+  mic_granted: "mic_granted",
+  audio_setup: "audio_setup",
+  finished: "finished"
+}
+
+async function grantedMicLabels() {
+  const mediaDevices = await navigator.mediaDevices.enumerateDevices();
+  return mediaDevices.filter(d => d.label && d.kind === "audioinput").map(d => d.label);
+}
+
+async function hasGrantedMicPermissions() {
+  const micLabels = await grantedMicLabels();
+  return micLabels.length > 0;
+}
+
+const TwoDEntryButton = (props) => (
+  <button {...props}>
+    Enter on this Screen
+  </button>
+);
+
+const GenericEntryButton = (props) => (
+  <button {...props}>
+    Enter in VR
+  </button>
+);
+
+const GearVREntryButton = (props) => (
+  <button {...props}>
+    Enter on GearVR
+  </button>
+);
+
+const DaydreamEntryButton = (props) => (
+  <button {...props}>
+    Enter on Daydream
+  </button>
+);
```

Comment:
yup I was thinking about how to work this in. I think "Using Google Chrome" makes a lot of sense since it jumps you to another browser on-screen and it's something users will understand. I don't think that we need to convey that they are going into Oculus Browser though, because conceptually they are probably unlikly to even know what Oculus Browser is, and since the next step after clicking is to put on their headset, they are unlikely to react to this as an "unexpected redirection to another browser" as they'd reasonably be expected to in the Chrome case.